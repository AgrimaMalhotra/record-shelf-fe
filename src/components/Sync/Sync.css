.sync-page {
    height: 80vh;
    display: flex;
    align-items: center;
    text-align: center;
    justify-content: center;
    font-size: 45px;
    font-family: Arial;
    width: 100%;
}

button {
    margin-top: 30px;
    font-size: 45px;
    font-family: Arial Black;
    background-color: #4C63AA;
    color: white;
    height: 80px;
    width: 250px;
    border-radius: 60px;
}