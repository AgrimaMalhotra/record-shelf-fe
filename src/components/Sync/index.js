import React from 'react';
import { useNavigate } from 'react-router-dom';
import './Sync.css';

const Sync = () => {
  const navigate = useNavigate();
  function handleClick() {
    navigate('songs');
  }
  return (
    <div className="sync-page">
      <div className="sync-page-wrapper">
        <p>
          :(( <br />
          seems a bit empty in here...
        </p>
        <button type="button" onClick={handleClick}>
          sync
        </button>
      </div>
    </div>
  );
};
export default Sync;
