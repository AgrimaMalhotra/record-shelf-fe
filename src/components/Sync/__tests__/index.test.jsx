import { fireEvent, render, screen } from '@testing-library/react';
import * as React from 'react';
import Sync from '..';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));
describe('Sync', () => {
  it('should render correctly ', () => {
    const { container } = render(<Sync />);
    expect(container).toMatchSnapshot();
  });
  it('should navigate to songs page on clicking sync button', async () => {
    render(<Sync />);
    const button = screen.getByTestId('sync-button');
    fireEvent.click(button);
    expect(mockNavigate).toHaveBeenCalledTimes(1);
  });
});
