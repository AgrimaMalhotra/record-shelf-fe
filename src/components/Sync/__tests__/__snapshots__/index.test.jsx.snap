// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sync should render correctly  1`] = `
<div>
  <div
    class="sync-page"
  >
    <div
      class="sync-page-wrapper"
    >
      <p>
        :(( 
        <br />
        seems a bit empty in here...
      </p>
      <button
        data-testid="sync-button"
        type="button"
      >
        sync
      </button>
    </div>
  </div>
</div>
`;
